src/main/kotlin/org/rust/ide/formatter/RustAlignmentStrategy.kt

```diff
@@ -0,0 +1,90 @@
+package org.rust.ide.formatter
+
+import com.intellij.formatting.Alignment
+import com.intellij.lang.ASTNode
+import com.intellij.psi.tree.IElementType
+import com.intellij.psi.tree.TokenSet
+
+interface RustAlignmentStrategy {
+    /**
+     * Requests current strategy for alignment to use for given child.
+     */
+    fun getAlignment(child: ASTNode, parent: ASTNode?, childCtx: RustFmtContext): Alignment?
```

Comment:
Just a random design note, not something that should be changed in this case. When you design an API based on callbacks, it may make sense to use parameter object pattern (`getAlignmemnt(ctx: AlignmentContext)`), because it allows to evolve the API better. Adding a new parameter to the interface breaks all the implementations, adding a field to a context does not.